src/ast/irbuilderbpf.cpp

```diff
@@ -607,14 +607,12 @@ CallInst *IRBuilderBPF::CreateProbeReadStr(Value *ctx,
   // int bpf_probe_read_str(void *dst, int size, const void *unsafe_ptr)
   FunctionType *probereadstr_func_type = FunctionType::get(
       getInt64Ty(), { dst->getType(), getInt32Ty(), src->getType() }, false);
-  PointerType *probereadstr_func_ptr_type = PointerType::get(
-      probereadstr_func_type, 0);
-  Constant *probereadstr_callee = ConstantExpr::getCast(
```

Comment:
Sorry, I might have missed it but what is this change?

Reply:
Oops, you're right this could have done with an explanation. It's to produce more useful error messages for the `bpf_probe_read_str` helper. A lot of other helpers are already using this `CreateHelperCall` utility function, and then rest probably should be switched to it in the future as well.

`bpf_probe_read_str` is a GPL-only helper function so I ended up getting a lot of error messages with this one in particular when playing around with different licenses.

Without this change:
```
# bpftrace -e 'config = {license = "poo"; } BEGIN { str(0); }' 
WARNING: Addrspace is not set
Attaching 1 probe...
ERROR: helper bpf_probe_read_str can only be used in GPL-compatible programs
```

With this change:
```
# bpftrace -e 'config = {license = "poo"; } BEGIN { str(0); }' 
WARNING: Addrspace is not set
Attaching 1 probe...
stdin:1:38-44: ERROR: helper bpf_probe_read_str can only be used in GPL-compatible programs
config = {license = "poo"; } BEGIN { str(0); }
                                     ~~~~~~
```